### PR TITLE
Be explicit that custom metadata interface implementations are not yet supported

### PR DESCRIPTION
**Pull request by ajcvickers**
See issue #5074

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore/Extensions/KeyExtensions.cs [hunk 1]
```diff
@@ -22,18 +22,6 @@ public static class KeyExtensions
...
-            if (referencingForeignKeyMetadata != null)
-            {
-                return referencingForeignKeyMetadata.ReferencingForeignKeys;
-            }
-
-            return key.DeclaringEntityType.Model.GetEntityTypes()
-                .SelectMany(e => e.GetDeclaredForeignKeys())
-                .Where(fk => fk.PrincipalKey == key);
-        }
+            => Check.NotNull(key, nameof(key)).AsKey().ReferencingForeignKeys ?? Enumerable.Empty<IForeignKey>();
```
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore/Extensions/KeyExtensions.cs]: Move the coalesce to .ReferencingForeignKeys

Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore/Extensions/KeyExtensions.cs]: All the FindReferencing* properties should be renamed to GetReferencing*
We decided that methods returning a collection that don't take parameters should be prefixed with Get. This and the ones for Property were missed.

File: src/Microsoft.EntityFrameworkCore/Metadata/Internal/MetadataExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    public static class MetadataExtensions
+    {
+        public static TConcrete AsConcreteType<TInterface, TConcrete>([NotNull] this TInterface @interface, [NotNull] string methodName)
+            where TConcrete : class
+        {
+            var concrete = @interface as TConcrete;
+            if (concrete == null)
+            {
+                throw new NotSupportedException(
+                    CoreStrings.CustomMetadata(methodName, typeof(TInterface).Name, @interface.GetType().Name));
```
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore/Metadata/Internal/MetadataExtensions.cs]: The exception message is specific for metadata interfaces. Either add a check so it is only thrown for metadata interfaces or rename the method to reflect this.

File: src/Microsoft.EntityFrameworkCore/Extensions/KeyExtensions.cs (hunk 1 above)
Comment (ajcvickers) [inline src/Microsoft.EntityFrameworkCore/Extensions/KeyExtensions.cs]: Will do.

File: src/Microsoft.EntityFrameworkCore/Metadata/Internal/MetadataExtensions.cs (hunk 2 above)
Comment (ajcvickers) [inline src/Microsoft.EntityFrameworkCore/Metadata/Internal/MetadataExtensions.cs]: Will rename,

**Comment by AndriySvyryd** [comment]
:shipit:

File: src/Microsoft.EntityFrameworkCore/Extensions/KeyExtensions.cs (hunk 1 above)
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore/Extensions/KeyExtensions.cs]: Same goes for FindContaining*

**Pull request merged**